evm/vm_state.py

```diff
@@ -97,14 +101,13 @@ def state_db(self, read_only=False):
         elif self.state_root != state.root_hash:
             self.set_state_root(state.root_hash)
 
-        self.access_logs.reads.update(state.db.access_logs.reads)
-        self.access_logs.writes.update(state.db.access_logs.writes)
+            self.access_logs.reads.update(state.db.access_logs.reads)
```

Comment:
Now I think `self.access_logs.reads.update(state.db.access_logs.writes)` should be in `elif self.state_root != state.root_hash` section while `self.access_logs.writes.update(state.db.access_logs.reads)` isn't.

Reply:
Seems like a typo? self.access_logs.**reads**.update(state.db.access_logs.**writes**)
I suppose `self.access_logs.reads.update(state.db.access_logs.reads)` should be updated regardless of whether it's read or write?